### PR TITLE
[WIP] replace source and model profiles with analytical queries

### PR DESCRIPTION
**Pull request by hamilton**
This is a draft PR that will have _all_ the changes needed to deprecate the existing Node-backed profile queries. I'll produce a work breakdown needed to move this PR out of draft -> ready for review as I get more done.

core
- [x] compose / unnest a nulls + cardinality + profile columns query to enable sorting
- [x] remove needlessly heavy nesting in profile components
- [x] migrate null percentage summary viz
- [x] migrate cardinality summary viz 
- [x] migrate numeric histogram summary viz
- [x] migrate numeric detail plots
- [x] migrate topK detail view (let's also ditch the existing topk in favor of the leaderboard version)
- [x] source: migrate preview table
- [x] model: migrate preview table
- [x] migrate in single point example
- [x] migrate time series summary viz + detail plots
- [x] get model changes to trigger the right update
- [ ] investigate why some timestamp columns don't fully work (@hamilton)
- [ ] remove all persistent and derived stores from the frontend (@ericpgreen2)
- [ ] when I change models, (1) move previous model to inactive and (2) move any inactive queries in the next model to active
- [ ] add sql to query key, make previous versions of query `disabled`
- [ ] when source changes, invalidate previous model queries 

finesse
- [x] restyle sparks
- [x] restyle numeric plots based on other branches
- [x] fix spacing issue
- [ ] fix timestamp rendering (@hamilton)
- [ ] remove buttons from numeric queries if there's no data (@hamilton)
- [ ] reorganize components and remove old profile components (@hamilton)

show-stoppers
- [ ] figure out how to speed up model profiling on keystroke. It currently takes one second before anything changes
- [ ] if I select from a table like the one made by [this source](https://storage.googleapis.com/rill-product.rilldata.com/example/nyc311-reduced.parquet), and I copy and paste a WHERE clause that is valid a few times, I can get the runtime to break.

### CODE REVIEW

File: web-local/src/lib/components/column-profile/ColumnSummaryMiniPlots.svelte [hunk 1]
```diff
@@ -45,13 +51,43 @@
     containerWidth > COLUMN_PROFILE_CONFIG.compactBreakpoint
       ? formatInteger
       : formatCompactInteger;
+
+  /**
+   * Get the null counts for this profile.
+   */
+  let nullCountQuery;
+  $: if ($runtimeStore?.instanceId)
```
Comment (AdityaHegde) [inline web-local/src/lib/components/column-profile/ColumnSummaryMiniPlots.svelte:59]: No need for the if here. The generated use hooks have the check internally already.

Comment (AdityaHegde) [reply web-local/src/lib/components/column-profile/ColumnSummaryMiniPlots.svelte:59]: Applies to all other such usage

Comment (hamilton) [reply web-local/src/lib/components/column-profile/ColumnSummaryMiniPlots.svelte:59]: got it, thanks. I believe this pattern might be used elsewhere by someone else, was copy-pasting in the interest of time.

**Review by ericpgreen2 — APPROVED** [review]
Approving, so we can merge into `main` and address remaining issues in smaller branches.

**Pull request merged**